frappe/core/doctype/scheduled_job_type/scheduled_job_type.json

```diff
@@ -92,6 +93,14 @@
   {
    "fieldname": "column_break_9",
    "fieldtype": "Column Break"
+  },
+  {
+   "default": "0",
+   "description": "Delays next execution to ease workload.",
+   "fieldname": "jitter",
+   "fieldtype": "Int",
+   "label": "Jitter",
+   "non_negative": 1
   }
```

Comment:
FYI this doctype gets created and destroyed automatically. It's unlikely that we will configure this in code/config :grimacing: 

We can just add some sane value for this and *maybe* make it configurable via site config (frappe.conf) if required.

Reply:
Yeah, I thought of setting it as hidden, but didn't know if users manually schedule jobs, so left it as it is.

In the PR it is [automatically set](https://github.com/frappe/frappe/pull/23132/files#diff-076c220e1d62b578715d96ada137fcba4da60638132563ec9754685f54bbe02cR157-R160) at the completion of a job.

By **configurable**, do you mean a setting that enables or disables jitter, or one that sets max jitter amount?

Either way, implementation will depend on a field to offset the next execution time.

Either that or some kind of smarter jitter needs to be applied that delays enqueue depending on how many jobs are clustered around the scheduler tick timestamp, but that involves evaluating `is_event_due` in batches, as opposed to one by one like how it's done now.

That seems like an overkill though.

If not that, I have missed something in which case, lmk.